eng/pipelines/templates/stages/archetype-net-release.yml

```diff
@@ -241,7 +241,7 @@ stages:
     dependsOn: Signing
     jobs:
     - job: PublishPackages
-      condition: and(eq(variables['SetDevVersion'], 'true'), eq(variables['System.TeamProject'], 'internal'))
+      condition: or(eq(variables['SetDevVersion'], 'true'), eq(variables['System.TeamProject'], 'internal'))
```

Comment:
We have to be careful with this fix as I think it might break in scenarios where there are multiple builds in internal where the SetDevVersion is not set to true. In those cases the build version will not be changed correctly and we will try to upload a package with the same version and thus fail.

Reply:
This is a revert to what it was previously though? So did that problem always exist?

Reply:
Looks like you made 2 independent changes originally. You need to also revert https://github.com/Azure/azure-sdk-for-net/pull/19669